packages/debug/src/browser/debug-tab-bar-decorator.ts

```diff
@@ -0,0 +1,62 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { Emitter, Event } from '@theia/core/lib/common/event';
+import { DebugSessionManager } from './debug-session-manager';
+import { TabBarDecorator } from '@theia/core/lib/browser/shell/tab-bar-decorator';
+import { FrontendApplication, FrontendApplicationContribution, Title, Widget } from '@theia/core/lib/browser';
+import { WidgetDecoration } from '@theia/core/lib/browser/widget-decoration';
+import { DisposableCollection } from '@theia/core/lib/common';
+
+export const DebugSessionManagerGetter = Symbol.for('DebugSessionManagerGetter');
+export type DebugSessionManagerGetter = () => DebugSessionManager;
+
+@injectable()
+export class DebugTabBarDecorator implements TabBarDecorator, FrontendApplicationContribution {
+    readonly id = 'theia-debug-tabbar-decorator';
+
+    protected readonly emitter = new Emitter<void>();
+    protected toDispose = new DisposableCollection();
+
+    @inject(DebugSessionManagerGetter)
+    protected readonly debugSessionManager: DebugSessionManagerGetter;
```

Comment:
Due to a circular dependency, it was not possible to directly inject `DebugSessionManager`. Thanks to @marechal-p for suggesting and helping me find a workaround on this.